compiler/src/compile.rs

```diff
@@ -337,28 +344,26 @@ impl Compiler {
     ) -> CompileResult<()> {
         self.symbol_table_stack.push(symbol_table);
 
-        let mut emitted_return = false;
-
-        for (i, statement) in body.iter().enumerate() {
-            let is_last = i == body.len() - 1;
-
+        for statement in body {
```

Comment:
alternatively could do `split_last()` and just handle the last one separately instead of editing the instruction stream

Reply:
it now looks a lot better, thanks!